src/hotspot/share/cds/dynamicArchive.cpp

```diff
@@ -344,24 +343,24 @@ void DynamicArchive::dump(const char* archive_name, TRAPS) {
   } else {
     // prevent multiple dumps.
     set_has_been_dumped_once();
-  }
-  ArchiveClassesAtExit = archive_name;
-  if (Arguments::init_shared_archive_paths()) {
-    dump();
-  } else {
-    ArchiveClassesAtExit = nullptr;
-    THROW_MSG(vmSymbols::java_lang_RuntimeException(),
+    ArchiveClassesAtExit = archive_name;
+    if (Arguments::init_shared_archive_paths()) {
+      dump(CHECK);
+    } else {
+      ArchiveClassesAtExit = nullptr;
+      THROW_MSG(vmSymbols::java_lang_RuntimeException(),
               "Could not setup SharedDynamicArchivePath");
-  }
-  // prevent do dynamic dump at exit.
-  ArchiveClassesAtExit = nullptr;
-  if (!Arguments::init_shared_archive_paths()) {
-    THROW_MSG(vmSymbols::java_lang_RuntimeException(),
+    }
+    // prevent do dynamic dump at exit.
+    ArchiveClassesAtExit = nullptr;
+    if (!Arguments::init_shared_archive_paths()) {
+      THROW_MSG(vmSymbols::java_lang_RuntimeException(),
              "Could not restore SharedDynamicArchivePath");
+    }
   }
 }
 
-void DynamicArchive::dump() {
+void DynamicArchive::dump(TRAPS) {
```

Comment:
TRAPS is no longer needed since this function doesn't throw exceptions anymore.

Reply:
Will revert to original version.